qiskit_experiments/calibration/update_library.py

```diff
@@ -0,0 +1,182 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A library of experiment calibrations."""
+
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import List, Tuple, Union
+import numpy as np
+
+from qiskit.circuit import Parameter
+from qiskit.pulse import ScheduleBlock
+
+from qiskit_experiments.experiment_data import ExperimentData
+from qiskit_experiments.calibration.backend_calibrations import BackendCalibrations
+from qiskit_experiments.calibration.calibrations import Calibrations
+from qiskit_experiments.calibration.parameter_value import ParameterValue
+from qiskit_experiments.calibration.exceptions import CalibrationError
+from qiskit_experiments.calibration.calibration_key_types import ParameterValueType
+
+
+class BaseUpdater(ABC):
+    """A base class to update calibrations."""
+
+    def __init__(self):
+        """Updaters are not meant to be instantiated."""
+        raise CalibrationError(
+            "Calibration updaters are not meant to be instantiated. The intended usage"
+            "is Updater.update(calibrations, exp_data, ...)."
+        )
+
+    @staticmethod
+    def _time_stamp(exp_data: ExperimentData) -> datetime:
+        """Helper method to extract the datetime."""
+        all_times = exp_data.completion_times.values()
+        if all_times:
+            return max(all_times)
+
+        return datetime.now()
+
+    @staticmethod
+    def _add_parameter_value(
```

Comment:
This function has the signature:
```python
 def _add_parameter_value(exp_data: ExperimentData, cal: Calibrations, ...)
```
(`ExperimentalData` and then `Calibrations`).

The `update()` functions on the derived classes have it the other way around.
Is there a reason for that. Its a bit of bike shedding but annoys me a bit when
trying to understand the code.

Reply:
Thanks for point this out. Fixed here: ce9b9fc